### PR TITLE
Clean logic to skip internal indexes.

### PR DESCRIPTION
**Pull request by GreggSchofield**
Looking forward to hearing your feedback!

### CODE REVIEW

**Comment by mrigger** [comment]
Thank you for the PR and sorry for the late response (I caught a cold). I believe that I left the original FIXME note since checking for the length of indexes being 2 does not provide any guarantees that we select only indexes that we created - the check might still erroneously include internal indexes with a name length of 2, and erroneously exclude indexes that we would want to select even when they exceed the length of 2 (e.g., `i11`). The PR does thus moves the workaround to the SQL side and does not really address the underlying issue.

I think a clean fix would look as follows: I found that Postgres uses a method in https://github.com/sqlancer/sqlancer/blob/61f61fe6c6e3bc64921b7dc296ef824e3ac2daca/src/sqlancer/sqlite3/gen/SQLite3Common.java#L33 to create an index of the specified format. It would be great to create a new class (e.g., named `sqlancer.common.DBMSCommon`), move the methods `createTableName`, `createColumnName` and `createIndexName` to there, and create a method `static boolean matchesIndexName(String indexName)` that returns true if the index name matches the format of the index name created by `createIndexName`. You could then use this newly-created method in the `getIndexes` method. Alternatively, it might be useful to check whether Postgres exposes this information somewhere and check for this.

Would you like to try that? Thanks again for the PR!

**Comment by GreggSchofield** [comment]
Thanks for your feedback Manuel! No worries I have been fighting a cold myself.

Ok I understand what you where trying to do now. Regarding your final comment about checking whether Postgres exposes information about its indexes I believe Postgres indexes are named using the 'pg_' prefix however I could not find any official documentation confirming this. I have implemented the solution you suggested by introducing the DBMSCommon utility class as thought this could possibly be re-used by other databases sqlancer supports (both currently and in the future).

All the best,

Gregg

File: src/sqlancer/common/DBMSCommon.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package sqlancer.common;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class DBMSCommon {
+
+    private static final Pattern sqlancerIndexPattern = Pattern.compile("^i\\d+");
```
Comment (mrigger) [inline src/sqlancer/common/DBMSCommon.java:8]: I think this should be `SQLANCER_INDEX_PATTERN` according to the Java naming convention.

**Review by mrigger — COMMENTED** [review]
Hi Gregg, thanks a lot for the changes, and I hope that you'll get better soon. The changes look good overall.

I just have a minor comment regarding the naming of a constant. If you use `mvn verify -DskipTests=true`, you see that Checkstyle complains about it. Typically, this would be caught by the Travis gate, but I just realized that Travis has a new pricing model and that the SQLancer repository is already out of credits - I think we'll migrate to GitHub actions soon.

It would be awesome if you could change the variable name, and we could merge the PR right after that. If it's not too much hassle, please use `git rebase -i` to fix this in the original commit that introduced the constant, so that we have a clean commit history (see https://github.com/sqlancer/sqlancer/blob/master/CONTRIBUTING.md#commit-history).

File: src/sqlancer/common/DBMSCommon.java (hunk 1 above)
Comment (GreggSchofield) [reply src/sqlancer/common/DBMSCommon.java]: Ahh yes thank you!

**Comment by mrigger** [comment]
Thanks for the PR again, I will merge it now. If you'd like to contribute more PRs, please follow the guidelines mentioned above:

> If it's not too much hassle, please use `git rebase -i` to fix this in the original commit that introduced the constant, so that we have a clean commit history (see https://github.com/sqlancer/sqlancer/blob/master/CONTRIBUTING.md#commit-history).

**Pull request merged**